src/napari_micromanager/_mda_handler.py

```diff
@@ -72,6 +72,7 @@ def __init__(self, mmcore: CMMCorePlus, viewer: napari.viewer.Viewer) -> None:
 
         # mapping of id -> (zarr.Array, temporary directory) for each layer created
         self._tmp_arrays: dict[str, tuple[zarr.Array, tempfile.TemporaryDirectory]] = {}
+        self._deck: deque = deque()
```

Comment:
good call. Also for the future we should go to just collections.deque (at least once napari-micro is python 3.9+ only)

https://docs.python.org/3/library/typing.html#typing.Deque

> Deprecated since version 3.9: [collections.deque](https://docs.python.org/3/library/collections.html#collections.deque) now supports subscripting ([]). See [PEP 585](https://peps.python.org/pep-0585/) and [Generic Alias Type](https://docs.python.org/3/library/stdtypes.html#types-genericalias).

Reply:
Yep, all the generic collections can be updated then.  3.8 end of life is Oct 2024